Combined fixes from git master:

From 9ffd34784b1fabe1569862b2d551d9cf06e46fa2 Mon Sep 17 00:00:00 2001
From: adonais <hua.andy@gmail.com>
Date: Tue, 2 Nov 2021 12:33:00 +0800
Subject: [PATCH 1/3] reserve exiv2 exception capture

(cherry picked from commit 3bd9571a0aff4611339eb57132e79b41af7acbb6)

* asturmlechner 2023-06-12: Drop unrelated translation fix from commit
---
 qimgv/sourcecontainers/documentinfo.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

From abe43e0dc7ee6d120c3e052562748d5a2a9f7d46 Mon Sep 17 00:00:00 2001
From: adonais <hua.andy@gmail.com>
Date: Wed, 3 Nov 2021 09:07:30 +0800
Subject: [PATCH 2/3] Fix exiv2 exception capture

(cherry picked from commit 08d5dea2074d055f102002d1a8f705e2ad2292f1)
---
 qimgv/sourcecontainers/documentinfo.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

From c4d9d5b08341af6709fe6263e90f4237c67eb14c Mon Sep 17 00:00:00 2001
From: DigitalDragon64 <digitaldragon64@gmail.com>
Date: Thu, 8 Jun 2023 15:21:26 +0200
Subject: [PATCH 3/3] Exiv2::AnyError was removed

(cherry picked from commit bff220e6e8851482e383b71fe778a870c96cd5f8)
---
 qimgv/sourcecontainers/documentinfo.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/qimgv/sourcecontainers/documentinfo.cpp b/qimgv/sourcecontainers/documentinfo.cpp
index 21979f01..9de2dd31 100644
--- a/qimgv/sourcecontainers/documentinfo.cpp
+++ b/qimgv/sourcecontainers/documentinfo.cpp
@@ -268,8 +268,9 @@ void DocumentInfo::loadExifTags() {
             exifTags.insert("UserComment", comment);
         }
     }
+    // No exception was caught, which may cause QT crash
     catch (Exiv2::Error& e) {
-        //std::cout << "Caught Exiv2 exception '" << e.what() << "'\n";
+        qDebug() << "Caught Exiv2 exception:\n" << e.what() << "\n";
         return;
     }
 #endif
